_test/test_sqw_pixels/test_mask.m

```diff
@@ -270,46 +222,139 @@ function test_mask_pixels_removes_pixels_given_in_mask_array(obj)
+            assertEqual(sum(masked_2d.data.npix(~obj.mask_array_2d)), 0);
+            %test_mask_sets_signal_in_masked_bins_to_zero
+            assertEqual(sum(masked_2d.data.s(~obj.mask_array_2d)), 0);
+            %test_mask_sets_error_in_masked_bins_to_zero
+            assertEqual(sum(masked_2d.data.e(~obj.mask_array_2d)), 0);
+
+            %test_mask_does_not_change_unmasked_bins_signal
+            assertEqual(masked_2d.data.s(obj.mask_array_2d), ...
+                obj.sqw_2d.data.s(obj.mask_array_2d));
+            % test_mask_does_not_change_unmasked_bins_error
+            assertEqual(masked_2d.data.s(obj.mask_array_2d), ...
+                obj.sqw_2d.data.s(obj.mask_array_2d));
+            % test_mask_does_not_change_unmasked_bins_npix
+            assertEqual(masked_2d.data.npix(obj.mask_array_2d), ...
+                obj.sqw_2d.data.npix(obj.mask_array_2d));
+
+            % test_num_pixels_has_been_reduced_by_correct_amount
+            expected_num_pix = sum(obj.sqw_2d.data.npix(obj.mask_array_2d));
+            assertEqual(masked_2d.pix.num_pixels, expected_num_pix);
+
+            % test_img_range_recalculated_after_mask
+
+            original_pix_range = obj.sqw_2d.pix.pix_range;
+            new_pix_range = masked_2d.pix.pix_range;
+            obj.masked_2d_pix_range = new_pix_range;
+
+            pix_range_diff = abs(original_pix_range - new_pix_range);
+            assertTrue(pix_range_diff(1) > 0.001);
+
+            assertElementsAlmostEqual(original_pix_range(2,:), new_pix_range(2,:), ...
+                'absolute', 0.001);
+            assertElementsAlmostEqual(original_pix_range(1,3:end), new_pix_range(1,3:end), ...
+                'absolute', 0.001);
+
+            if isempty(obj.masked_2d_paged_cache)
+                [~, masked_2d_paged,clPageConfig] = ...
+                    obj.get_paged_sqw(obj.sqw_2d_file_path, obj.mask_array_2d);
+                obj.masked_2d_paged_cache = masked_2d_paged;
+            else
+                masked_2d_paged = obj.masked_2d_paged_cache;
+            end
+
+            % test_paged_and_non_paged_sqw_have_equivalent_pixels_after_mask
+            raw_paged_pix = concatenate_pixel_pages(masked_2d_paged.pix);
+            assertEqual(raw_paged_pix, masked_2d.pix.data);
+        end
+
+
+        function test_mask_random_fraction_removes_perc_of_pix_in_mem(obj)
+            sqw_obj = sqw(obj.sqw_2d_file_path,'file_backed',false);
 
             frac_to_keep = 0.8;
             new_sqw = mask_random_fraction_pixels(sqw_obj, frac_to_keep);
 
             expected_num_pix = round(frac_to_keep*sqw_obj.pix.num_pixels);
             assertEqual(new_sqw.pix.num_pixels, expected_num_pix);
+
+            test_mask.check_filebacked_signal_averages(new_sqw);
+
+        end
+        function test_mask_random_retains_correct_number_of_pix_on_file(obj)
+            skipTest('Re #1081 The test illustrates broken filebacked masking. Should be enabled as part of fixing the ticket')
```

Comment:
Why is this new test disabled when this is supposed to fix this issue?

Reply:
This test do not fix the issue. It demonstrates it. Can it be interpreted differently?